backend/capellacollab/projects/capellamodels/modelsources/git/crud.py

```diff
@@ -33,58 +36,65 @@ def get_gitmodel_by_id(db: Session, id: int) -> DB_GitModel:
     return db.query(DB_GitModel).filter(DB_GitModel.id == id).first()
 
 
-def make_gitmodel_primary(db: Session, id: int) -> DB_GitModel:
-    primary_model = get_primary_gitmodel_of_capellamodels(db, id)
-    if primary_model:
-        primary_model.primary = False
-        db.add(primary_model)
+def make_git_model_primary(
+    db: Session, capella_model_id: int, git_model_id: int
+) -> DB_GitModel:
+    primary_model = get_primary_gitmodel_of_capellamodel(db, capella_model_id)
+    primary_model.primary = False
 
-    new_primary_model = get_gitmodel_by_id(db, id)
-    new_primary_model.primary = True
+    patch_git_model = get_gitmodel_by_id(db, git_model_id)
+    patch_git_model.primary = True
 
-    db.add(new_primary_model)
     db.commit()
-    db.refresh(new_primary_model)
-    return new_primary_model
+    return patch_git_model
 
 
-def add_gitmodel_to_capellamodel(
-    db: Session, capellamodel_id: int, model: PostGitModel
+def delete_model_from_repository(
+    db: Session, capellamodel_id: int, model_id: int
 ):
-    if len(get_gitmodels_of_capellamodels(db, capellamodel_id)):
+    db.query(DB_GitModel).filter(DB_GitModel.id == model_id).filter(
+        DB_GitModel.model_id == capellamodel_id
```

Comment:
Why do you need to filter the model of the git model? The behaviour here is to fail silently in case there is no model matching, I’m not sure that’s the desired behaviour.

Reply:
I haven't actually made any changes here, but somehow (I suspect due to changing the order of some functions) it shows up here. But yes, you are right that this is probably not the desired behavior. Since the function is not needed at the moment and is just a simple crud function, I removed it so we don't just accidentally use it at a later point.